modules/cli/apply/apply.go

```diff
@@ -167,10 +174,46 @@ func readBundle(ctx context.Context, name, baseDir string, opts *Options) (*bund
 	})
 }
 
+// createName uses the bundle name + the path to the bundle to create a unique name
+// name: the gitrepo name, passed to 'fleet apply' on the cli
+// basedir: the path from the walk func in Dir, []baseDirs
 func createName(name, baseDir string) string {
-	path := strings.ToLower(filepath.Join(name, baseDir))
-	path = disallowedChars.ReplaceAllString(path, "-")
-	return name2.Limit(multiDash.ReplaceAllString(path, "-"), 63)
+	needHex := false
+
+	str := filepath.Join(name, baseDir)
+	str = strings.ReplaceAll(str, "/", "-")
+
+	// avoid collision from different case
+	if str != strings.ToLower(str) {
+		needHex = true
+	}
+
+	// avoid collision from disallowed characters
+	if disallowedChars.MatchString(str) {
+		needHex = true
+	}
+
+	if needHex {
+		// append checksum before cleaning up the string
+		str = fmt.Sprintf("%s-%s", str, name2.Hex(str, 8))
+	}
+
+	// clean up new name
+	str = strings.ToLower(str)
+	str = disallowedChars.ReplaceAllLiteralString(str, "-")
+	str = multiDash.ReplaceAllString(str, "-")
+	str = strings.TrimLeft(str, "-")
+	str = strings.TrimRight(str, "-")
+
+	// shorten name to 53 characters, the limit for helm release names
```

Comment:
I would extend the comment that the Limit function of wrangler adds a checksum to prevent collisions itself.